### ISSUE TITLE
2020-3-26

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#248]
381.3.TCNN model could not be trained at all, since gradients will vanish or blow up during back propagation through a deep network; (2020-3-7)
382.4.the program representation serves as a pretraining method, the training and CV errors decrease drastically; (2020-3-7)
383.5.The fact that unsupervised pretraining improves supervised learning; As pretraining explores underlying data features unsupervised, it gives a much more meaningful initialization of parameters; deep neural networks can be trained much faster and more effectively; (2020-3-7)
384.6.TCNN is better than SVM; (2020-3-7)
385.7.Sum up: (1) evaluate the learned representations empirically by nearest neighbor querying and k-means clustering; (2) learned representations are greatly beneficial for supervised learning; (2020-3-7)
386.VI.LOOKING FORWARD TO THE FUTURE; 
387.A.Different Perspectives for Program Modeling; 
388.1.we treat a program as a tree; hypothesize that structural information is important; (2020-3-12)
389.2.treating a program as a sequence of statements; neighboring information are captured and local patterns can be extracted; (2020-3-12)
